src/MaterialEditor.Base/UI/UI.cs

```diff
@@ -352,9 +371,17 @@ protected void PopulateList(GameObject go, object data, string filter = "")
                     string propertyName = property.Key;
                     if (Instance.CheckBlacklist(materialName, propertyName)) continue;
 
-                    if (property.Value.Type == ShaderPropertyType.Texture)
+                    bool showProperty = filterListProperties.Count == 0;
+                    foreach (string filterWord in filterListProperties)
+                        if (WildCardSearch(propertyName, filterWord))
+                        {
+                            showProperty = true;
+                            break;
+                        }
+
+                    if (showProperty && mat.HasProperty($"_{propertyName}"))
                     {
-                        if (mat.HasProperty($"_{propertyName}"))
+                        if (property.Value.Type == ShaderPropertyType.Texture)
```

Comment:
Why was the order of those checks changed? The `.HasProperty` call is much more expensive than comparing the Type which is an enum and should only be done if necessary.

Also, the Keyword type is not filtered, is this intentional?

Reply:
ah, that explains why the if check was done multiple time instead of once at the top. I will change it back.

The keyword type I just forgot about tbh, so I'll add that too